onnxruntime/core/providers/js/operators/gather.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/js/js_kernel.h"
+
+#include "gather.h"
+
+namespace onnxruntime {
+namespace js {
+
+using AllSupportedSize =
+    TypeList<
+        float,
+        double,
+        int64_t,
+        uint64_t,
+        int32_t,
+        uint32_t>;
+
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(
+    Gather,
+    kOnnxDomain,
+    1,
+    10,
+    kJsExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .TypeConstraint("T", BuildKernelDefConstraintsFromTypeList<AllSupportedSize>())
+        .TypeConstraint("Tind", BuildKernelDefConstraintsFromTypeList<TypeList<int32_t, int64_t>>()),
```

Comment:
Yeah. Most likely it will be even faster. Will update the code to have the second input on CPU tomorrow

Reply:
I just want to know. Not necessarily requiring changes.

Reply:
I think in most cases inputs[1] would be a constant. But technically it can be an output from other GPU node. I just did it that way because it seemed easier to implement